solr/core/src/java/org/apache/solr/highlight/UnifiedSolrHighlighter.java

```diff
@@ -314,6 +343,26 @@ protected PassageFormatter getFormatter(String fieldName) {
       return new DefaultPassageFormatter(preTag, postTag, ellipsis, "html".equals(encoder));
     }
 
+    @Override
+    protected Comparator<Passage> getPassageSortComparator(String fieldName) {
+      String value = params.get(HighlightParams.PASSAGE_SORT, PassageSort.START_OFFSET.key);
+
+      PassageSort passageSort = PassageSort.parse(value);
+
+      switch (passageSort) {
+        case START_OFFSET:
+          return Comparator.comparingInt(Passage::getStartOffset);
```

Comment:
```suggestion
      switch (passageSort) {
        case "startOffset:
          return Comparator.comparingInt(Passage::getStartOffset);
```

Is this the direction you were aiming for? Are you suggesting that we avoid creating unnecessary enum class and use string literals instead?

Reply:
Yes, because the enum isn't providing much of any value.  Isn't a useful "type".

Reply:
It has been reflected in commit d5618e5.